handler/src/main/java/com/networknt/handler/config/PathChain.java

```diff
@@ -32,11 +39,37 @@ public String getMethod() {
         return method;
     }
 
-    public void setMethod(String method) throws Exception {
-        if (NetUtils.METHODS.contains(method.toUpperCase())) {
-            this.method = method;
+    public void setMethod(String method) { this.method = method; }
+
+    @Override
+    public String toString() {
+        if(path != null) {
+            return path + "@" + method + " → " + exec;
         } else {
-            throw new Exception("Invalid HTTP method provided: " + method);
+            return source + "() → " + exec;
+        }
+    }
+    /**
+     * Validate the settings and raise Exception on error.
+     * The origin is used to help locate problems.
+     */
+    public void validate(String origin) {
+        List<String> problems = new ArrayList<>();
+        if(source == null) {
+            if(path == null) {
+                problems.add("You must specify either path or source");
+            } else if(method == null) {
+                problems.add("You must specify method along with path: " + path);
+            }
+        } else {
+            if(path != null) problems.add("Conflicting source: " + source + " and path: " + path);
+            if(path != null) problems.add("Conflicting source: " + source + " and method: " + method);
```

Comment:
No?

I'm trying to run all the check and report all the problems in a single validation call so the deployer doesn't have to go through a build/deploy/run cycle to catch each problem separately. however, the second `path != null` should be `method != null`. I have written a unit test for that case and fixed it locally. Will push shortly.

Reply:
The second `path != null` to `method != null` is what i meant by "check should be method" 😆 

Thanks!

Reply:
The second check there should be `method != null` so that's a bug. It's fixed locally with unit tests and will be pushed in a bit.

I don't think the code would become more readable by extracting those two lines into s method:
```
addProblemsConflictWithSource(problems, source, path, "path");
addProblemsConflictWithSource(problems, source, method, "method");
```